### ISSUE TITLE
dist client

### ISSUE THREAD

**Issue by billy-tutch** [HabitRPG/habitica#9043]
{"success":false,"error":"Error","message":"ENOENT: no such file or directory, stat 'C:\\Users\\Administrator\\Desktop\\habitica\\dist-client\\index.html'"}

What's missing, i don't have a dist-client folder

**Comment by paglias** [comment]
When are you getting this message?

**Comment by billy-tutch** [comment]
When loading the localhost.

NPM Start runs fine, installation went through without any errors. I don't have a dist-client folder anywhere?

**Comment by paglias** [comment]
I see, we've to update the docs. Basically you can run the client in a separate process running `npm run client:dev` and find it at `localhost:8080`, with the api at `localhost:3000` with automatic rebuilding when files change or build the client only once with `npm run client:build` and access it alongside the api at `localhost:3000` but it takes awhile to build it everytime and it's only meant for production usage

**Comment by luanmuniz** [comment]
@paglias Closing the issue before updating the docs! :(
Just lost an hour looking for this

How can we update the docs?

**Comment by paglias** [comment]
@luanmuniz the docs are up to date here http://habitica.wikia.com/wiki/Setting_up_Habitica_Locally#Run_Habitica

**Comment by luanmuniz** [comment]
I had to run `npm run client:build`, not `npm run client:dev` to make it work. And actually keep rerunning it with `npm start` everytime i did a change =/

**Comment by paglias** [comment]
@luanmuniz Try this:

- Open two terminal windows
- Into one run `npm start`
- Into the other `npm run client:dev`
- The second command will take some time for the initial build
- Then you'll have the website available at `localhost:8080` and the api at `localhost:3000`, but you can ignore that as all the api requests to `localhost:8080/api` are proxies to the api server at localhost:3000
- Any change made to /website files will not require any new command to be run, if you change a client file the site will reload automatically with the new changes, if you make a change to /website/common/locales instead you'll have to reload the site to see the changes

**Comment by luanmuniz** [comment]
@paglias Here is what happen:

If i run `npm start` and `npm run client:dev`, there is no `dist-client` folder. So when i access `localhost:3000` to access the website, i receive a message 

```
2017-12-06T12:48:50.474Z - warn: Error: ENOENT: no such file or directory, stat '/vagrant/dist-client/index.html'
    at Error (native)
{ method: 'GET',
  originalUrl: '/',
  headers:
   { host: 'localhost:3000',
     connection: 'keep-alive',
     'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36',
     'upgrade-insecure-requests': '1',
     accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8',
     dnt: '1',
     'accept-encoding': 'gzip, deflate, br',
     'accept-language': 'en-US,en;q=0.9,pt-BR;q=0.8,pt;q=0.7,es;q=0.6' },
  body: {},
  httpCode: 404,
  isHandledError: true,
  fullError:
   { errno: -2,
     code: 'ENOENT',
     syscall: 'stat',
     path: '/vagrant/dist-client/index.html',
     expose: false,
     statusCode: 404,
     status: 404 } }
```

If i run `npm run client:build` the `dist-client` folder will be created and i can access the website. But then i need to run `client:build` everytime i made a modification

Even after i have the `dist-client` folder, if i try to run `npm start` in one terminal and `npm run client:dev` in another, the folder will not be updated.

Basically: `npm run client;dev` is not doing what's suppose to do.

Its easy to replicate this:
- Clone the repository
- RUN `cd habitica`
- RUN `cp Vagrantfile.example Vagrantfile`
- RUN `cp config.json.example config.json`
- RUN `vagrant up`
- RUN `vagrant ssh`
- RUN `npm install`
- RUN `gulp build`
- RUN `npm start`
- Open another terminal
- RUN `cd habitica`
- RUN `vagrant ssh`
- RUN `npm run client:dev`

With these steps you will see that if i try to open `localhost:3000` i have the error above. I manage to reproduce this multiple times.

Obs: localhost:8080 is not open on the vagrantfile.

**Comment by paglias** [comment]
When running npm run client:dev the site is accessible at localhost:8080 not at localhost:3000

**Comment by paglias** [comment]
@SabreCat @TheHollidayInn @Alys do you know anything about 8080 not being available on Vagrant?

**Comment by Alys** [comment]
It looks like https://github.com/HabitRPG/habitica/blob/develop/vagrant_scripts/vagrant.sh hasn't been updated for the new method of starting the local servers.

**Comment by luanmuniz** [comment]
@paglias @Alys Anything that i can do?

**Comment by paglias** [comment]
@luanmuniz if you want to open a PR that fixes those vagrant script that would be great, Otherwise I'd suggest to follow the instructions here http://habitica.wikia.com/wiki/Setting_up_Habitica_Locally for you operative system

**Comment by BradLugo** [comment]
@paglias It appears that the Dockerfile has the same issue. I can make a PR for that but I need to know how y'all want to handle this with containers. Idiomatically, each container should only be responsible for one function. I can make multiple Dockerfiles and have the docker compose run both of them if y'all want

**Comment by TheHollidayInn** [comment]
A Docker file for client and one for server would be good. The production docker file should be good though.

**Comment by paglias** [comment]
I don't think the production docker works locally because it uses the compiled JS/CSS files, serving them from the main server, while locally it should spawn two different servers: `npm start` on port 3000 for the server and, `npm run client:dev` on port 8080 for the client. So if we want Docker for local development we'll need a different setup

**Comment by BradLugo** [comment]
I'm working on getting the Dockerfile fixed. What's the intention of Dockerfile-Production? I didn't see any documentation on it. I can fix that one too but I'll leave it alone if the expectation is for the admins to manage that for production use

**Comment by paglias** [comment]
Yes, the Docker production file should remain the same as now, it's used to deploy the habitica.com website

**Comment by BradLugo** [comment]
@Alys It looks like the script you referenced early just installs dependencies, not necessarily runs the code. I'm going to submit a PR for the Vagrant.example in a minute but it looks like the only thing that needs to be changed is the port forwarding. Can you confirm this?

**Comment by Alys** [comment]
@BradLugo That sounds reasonable. http://habitica.wikia.com/wiki/Setting_up_Habitica_Locally should be edited too to explain that a different port than 8080 might be selected (in the same way it already explains that vagrant might select a different port than 3000)

**Comment by BradLugo** [comment]
Once #9732 is merged in, I'll update the docs to reflect the change
